apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -516,12 +521,18 @@ export default class InstallManager {
         return;
       }
     } else {
-      console.log(`Deleting the NPM cache folder`);
-      // This is faster and more thorough than "npm cache clean"
-      this._asyncRecycler.moveFolder(this._rushConfiguration.npmCacheFolder);
+      if (this._rushConfiguration.packageManager === 'npm') {
+        console.log(`Deleting the packages-cache folder`);
+        // This is faster and more thorough than "npm cache clean"
+        this._asyncRecycler.moveFolder(this._rushConfiguration.npmCacheFolder);
+
+        console.log(`Deleting the "packages-tmp" folder`);
+        this._asyncRecycler.moveFolder(this._rushConfiguration.npmTmpFolder);
 
-      console.log(`Deleting the "npm-tmp" folder`);
-      this._asyncRecycler.moveFolder(this._rushConfiguration.npmTmpFolder);
+      } else if (installType === InstallType.UnsafePurge) {
+        console.log(`Deleting the PNPM store folder`);
+        this._asyncRecycler.moveFolder(this._rushConfiguration.npmCacheFolder);
+      }
```

Comment:
Should this go in the subclasses?

Reply:
This class was simple enough that I felt subclassing was unnecessary

---
In reply to: [151786113](https://github.com/Microsoft/web-build-tools/pull/427#discussion_r151786113) [](ancestors = 151786113)